packages/pds/src/api/com/atproto/server/createAccount.ts

```diff
@@ -137,3 +132,29 @@ export default function (server: Server, ctx: AppContext) {
     }
   })
 }
+
+export const ensureCodeIsAvailable = async (
+  db: Database,
+  inviteCode: string,
+  withLock = false,
+): Promise<void> => {
+  const invite = await db.db
+    .selectFrom('invite_code')
+    .selectAll()
+    .where('code', '=', inviteCode)
+    .if(withLock && db.dialect === 'pg', (qb) => qb.forUpdate().skipLocked())
```

Comment:
We have a skip-locked here but no retrying mechanism the way you originally setup with repo_root— just wanted to make sure that's intentional.  Maybe the skip isn't as necessary given that a sanity check has already been done by the time a lock is taken.

Reply:
Yeah I wanted to keep complexity low & this tx is so short-lived that this should really only be hit if someone is trying to abuse & use the same invite code twice in rapid succession. I'm fine with throwing them errors 🤷‍♂️

Reply:
I suppose there's a chance two different folks have the same invite w/ multiple uses and try it at the same time, but chances there are negligible and retrying still works just fine 👍